mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -529,22 +535,29 @@ private boolean userHasLookupRightsOn(Mailbox mailbox, MailboxSession session) t
     public void deleteMailbox(final MailboxPath mailboxPath, final MailboxSession session) throws MailboxException {
         LOGGER.info("deleteMailbox {}", mailboxPath);
         assertIsOwner(session, mailboxPath);
-        final MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
+        MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
+        MessageMapper messageMapper = mailboxSessionMapperFactory.getMessageMapper(session);
 
-        Mailbox mailbox = mapper.execute((Mapper.Transaction<Mailbox>) () -> {
-            final Mailbox mailbox1 = mapper.findMailboxByPath(mailboxPath);
+        mapper.execute((Mapper.Transaction<Mailbox>) () -> {
+            Mailbox mailbox1 = mapper.findMailboxByPath(mailboxPath);
             if (mailbox1 == null) {
                 throw new MailboxNotFoundException(mailboxPath);
             }
 
+            QuotaRoot quotaRoot = quotaRootResolver.getQuotaRoot(mailboxPath);
+            long messageCount = messageMapper.countMessagesInMailbox(mailbox1);
+            long totalSize = Iterators.toStream(messageMapper.findInMailbox(mailbox1, MessageRange.all(), MessageMapper.FetchType.Metadata, -1))
```

Comment:
Better idea?

Reply:
BTW the cost of that operation is still acceptable IMO:
 - We read fast access metadata. I estimate the cost of that operation to be maximum 10s (basing myself on previous "select" work, which did similar improvements.

I'd rather not have yet another projection for this... For consistency, and migration reasons. Let see how it behaves before trying over-optimizing it. WDYT?

Reply:
you can't just tell a long it can be without : the backend type, the server specification, the mailbox size, etc. So it can be 10s, 10ms or 10h.
You confuse *implementation* with *api*
Let's say you add a : `computeMailboxSize()` or `computeMessageSetSize(MessageSet)` method to MessageMapper. You can very well have a default implementation doing exactly what you did.
But you can then optimize a specific backend for this query (and not only with a cassandra projection). It can use Event Sourcing, It can use SQL `SELECT sum(size) WHERE MAILBOX=x` (with is way more efficient than this `findInMailbox` call), etc.